src/uu/env/src/env.rs

```diff
@@ -106,8 +106,15 @@ struct Options<'a> {
 fn print_env(line_ending: LineEnding) {
     let stdout_raw = io::stdout();
     let mut stdout = stdout_raw.lock();
-    for (n, v) in env::vars() {
-        write!(stdout, "{n}={v}{line_ending}").unwrap();
+    // vars_os in case we get some invalid strings
+    for (n, v) in env::vars_os() {
+        write!(
+            stdout,
+            "{}={}{line_ending}",
+            n.to_string_lossy(),
+            v.to_string_lossy()
```

Comment:
I think that's not correct because GNU `env` doesn't output a lossy string:
```
$ FOO=$'\xFF\xFE' env | hexdump -C
46 4f 4f 3d ff fe  // I omitted everything else
```
I don't know about the key, I wasn't able to create an env var whose key contains a non-utf8 char.